MoltenVK/MoltenVK/Commands/MVKCommandPipelineStateFactoryShaderSource.h

```diff
@@ -402,5 +402,20 @@ kernel void accumulateOcclusionQueryResults(device VisibilityBuffer& dest [[buff
     if (dest.count < max(oldDestCount, src.count)) { dest.countHigh++; }                                        \n\
 }                                                                                                               \n\
                                                                                                                 \n\
+kernel void remapVertices(device const uint* vertexIndices [[ buffer(0) ]],                                     \n\
+                          device uint* remappedVertices [[ buffer(1) ]],                                        \n\
+                          uint numVertices [[ thread_position_in_grid ]]) {                                     \n\
+    uint vertexCount = numVertices;                                                                             \n\
+    for (uint vertexIndex = 0; vertexIndex < vertexCount; ++vertexIndex) {                                      \n\
+        uint index = vertexIndices[vertexIndex];                                                                \n\
+        if (index % 2 == 0) {                                                                                   \n\
+            index = index / 2;                                                                                  \n\
+        } else {                                                                                                \n\
+            index = vertexCount - 1 - index / 2;                                                                \n\
+        }                                                                                                       \n\
+                                                                                                                \n\
+        remappedVertices[vertexIndex] = index;                                                                  \n\
+    }                                                                                                           \n\
+}                                                                                                               \n\
```

Comment:
I understand this is a WIP, but I'd like to understand how this shader works.

`thread_position_in_grid` will be a unique index for each instance, and I assume, one for each vertex that is to be remapped? 

I'm not sure you want to be looping here, as the higher instances will tromp on the lower instances. If `thread_position_in_grid` does represent the original index, then the calculation of `index` could just be done once per shader instance.